components/dashboard/DashboardAssignments.js

```diff
@@ -23,27 +23,59 @@ class DashboardAssignments extends Component {
     const assignmentFilters = [
       { name: 'Teams', id: 'teams' },
       { name: 'Favorites', id: 'favorites' },
-      { name: 'All', id: 'all' }
+      { name: 'All', id: 'userContributions' }
     ]
-
     let teamAssignments = sortBy(prop('team_priority'), assignments).map(task => {
       return {
         priority: task.team_priority ? `team priority: ${task.team_priority}` : task.priority,
         name: task.name,
         assigned_by: task.team_name,
-        campaign_hashtag: task.campaign_hashtag
+        campaign_hashtag: task.campaign_hashtag,
+        source: 'ASSIGNMENT'
+      }
+    })
+    let userFavorites = favorites.map(favorite => {
+      return {
+        campaign_hashtag: favorite.campaign_hashtag,
+        campaign_id: favorite.campaign_id,
+        name: favorite.name,
+        priority: favorite.priority,
+        tasker_id: favorite.tasker_id,
+        tm_id: favorite.tm_id,
+        source: 'FAVORITES'
+      }
+    })
+    let userContributions = all.map(campaign => {
+      return {
+        author: campaign.author,
+        campaign_hashtag: campaign.campaign_hashtag,
+        changeset_comment: campaign.changeset_comment,
+        created_at: campaign.created_at,
+        description: campaign.description,
+        done: campaign.done,
+        geometry: campaign.geometry,
+        id: campaign.id,
+        instructions: campaign.instructions,
+        name: campaign.name,
+        priority: campaign.priority,
+        status: campaign.status,
+        tasker_id: campaign.tasker_id,
+        tm_id: campaign.tm_id,
+        updated_at: campaign.updated_at,
+        validated: campaign.validated,
+        source: 'CONTRIBUTION'
       }
     })
     const allCampaigns = {
-      favorites: sortBy(prop('priority'), favorites),
+      favorites: sortBy(prop('priority'), userFavorites),
       teams: teamAssignments,
-      all
+      userContributions
     }
+    console.log(userContributions)
     const assignmentsTable = allCampaigns[this.state.assignmentsFilter].map((assignment) => {
```

Comment:
@alyssadelaine @kamicut Something strange is happening with the tabs and the mapping function here for the `assigned_by` conditional. If you check out your All Campaigns table on your dashboard, as you switch back and forth between the 'Favorites' and 'All' tabs, you'll see that your favorites continue to be mapped/duplicated in both tabs.